src/component/Chart.jsx

```diff
@@ -317,11 +323,34 @@ class Chart extends Component {
             // map rows
             .map((row: ChartRow, index: number): List<Object> => {
                 // then dimensions
-                return this.dimensions
-                    .map((dimension: Map, dimensionName: string): Map => {
+                return  this.dimensions
+                    .reduce((
+                        flattenedDimensions: Map,
+                        dimension: Map,
+                        dimensionName: string
+                    ): Map => {
                         const appliedDimension = applyDimension(dimension, inheritedProps);
-                        return getters.get(dimensionName)(index, appliedDimension.get('columnName'));
-                    })
+                        const columns = List([].concat(appliedDimension.get('columnName')));
+
+                        const newDimensions = columns.reduce((
```

Comment:
Can you put some comments here documenting the intent of this part? Trying to keep the business logic in Chart rememberable.